beacon-chain/db/kv/state.go

```diff
@@ -383,6 +384,46 @@ func (k *Store) HighestSlotStates(ctx context.Context) ([]*state.BeaconState, er
 	return states, nil
 }
 
+// statesAtSlotBitfieldIndex retrieves the states in DB given the input index. The index represents
+// the position of the slot bitfield the saved state maps to.
+func (k *Store) statesAtSlotBitfieldIndex(ctx context.Context, tx *bolt.Tx, index uint64) ([]*state.BeaconState, error) {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.statesAtSlotBitfieldIndex")
+	defer span.End()
+
+	highestSlot := index - 1
```

Comment:
what if index is 0, this overflows

Reply:
addressed the line below :) 

`highestSlot = uint64(math.Max(0, float64(highestSlot)))`

Reply:
wouldn't 
```
math.Max(0, float64(highestSlot))
```
return the larger number which will be `MaxFloat64` ?

Reply:
Nice catch! It's now taking int instead of uint64, also verified via a test. Thank you :)